model/src/main/java/org/openremote/model/security/UserResource.java

```diff
@@ -40,14 +38,6 @@
 @Path("user")
 public interface UserResource {
 
-    @GET
-    @Path("{realm}/roles")
-    @Produces(APPLICATION_JSON)
-    @SuppressWarnings("unusable-by-js")
-    @RolesAllowed(Constants.READ_ADMIN_ROLE)
-    @Operation(operationId = "getRoles", summary = "Retrieve roles for a realm")
-    Role[] getRoles(@BeanParam RequestParams requestParams, @PathParam("realm") String realm);
-
```

Comment:
Don't we want to deprecate this endpoint instead of removing it?
Especially since it was calling `getClientRoles` in `UserResourceImpl` anyway.
So other than "cleaning up" it doesn't cause any maintenance overhead.

Alternatively, we could include this a minor update like 1.4.0, and notify users of the API change.

Reply:
Given we have static typing of the frontend model and this endpoint is specifically only for our own `openremote` client there shouldn't be outside usage and any that there is (in custom projects for example) will quickly generate a compilation error and then a simple method rename is all that is needed.

Even if this endpoint was left there would be a breaking change as the return type is now different.

The UserResource is really intended for use by the manager which is updated in lockstep, deciding if this is an announced breaking change is something we should decide.